enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+* Customisation of bootstrap manifests via control plane provider configuration
+
+### Workflow Description
+
+#### Opinionated installer generated infrastructure definitions
+
+The installer binary will be updated to transform the existing install config into Cluster API resources.
+
+During the `manifests` phase, the installer will generate the following Cluster API resources into an `infrastructure-manifests` folder:
+* Cluster
+* OpenShiftControlPlane
+* InfrastructureCluster (e.g. `AWSCluster`)
+* InfrastructureMachineTemplate (e.g. `AWSMachineTemplate`)
+
+Additionally supplemental resources will be required:
+* Namespace (each cluster will be created in a unique namespace for isolation)
+* Infrastructure Credentials secret (to allow the Cluster API control plane to authenticate to the cloud provider).
+
+The installer will then diverge from the current path, which is to execute `ignition-configs` and then `cluster` phases.
+
+The `ignition-configs` phase will become a terminal phase and will be used by the bootstrap provider to generate the ignition configs for the bootstrap node, control plane nodes and worker nodes.
+No changes are expected to the `ignition-configs` phase and existing UPI workflows should continue to work.
+
+The `cluster` phase will now skip the `ignition-configs` phase and will instead apply the Cluster API resources generated in the `manifests` phase to the Cluster API control plane.
+
+The installer will directly apply the Cluster API resources to the Cluster API control plane.
+If any modifications are made between the `manifests` and `cluster` phases, the installer will not overwrite these changes.
+This allows the user to tweak the configuration generated by the installer should they wish to do so.
+
+The installer will also create an `install-state` secret in the cluster namespace containing the current `.openshift_install_state.json` file.
+This will then be consumed later by the bootstrap provider to generation ignition.
+This secret will be referenced in the `OpenShiftControlPlane` spec.
+
+To allow the user to customise manifests, the installer will take all manifests from the `manifests` and `openshift` folders and wrap them into secrets to be applied to the cluster namespace.
+Each secret will be annotated to indicate that it should be included in the ignition generation phase, and to identify whether it was a `manifest` file or `openshift` file.
+A selector on the `OpenShiftControlPlane` spec will be used to identify which secrets should be included in the ignition generation phase.
+
+The bootstrap provider will read the `OpenShiftControlPlane` spec to determine the install state and manifests required to complete the bootstrap ignition generation and will reconstruct the required structure for the installer to complete the `ignition-configs` phase in cluster.
+
+Once all resources are applied, the installer will watch the Cluster API control plane resource status to determine when the cluster is ready.
```

Comment:
At each phase we know which resources need to be checked. So, the installer can do the following:
* Watch the status of the Cluster object, until it reports that it is provisioned
* While watching the `Cluster`, until `Cluster.Status.InfrastructureReady  true`, watch InfraCluster, eg AWSCluster for status
* Once `Cluster.Status.InfrastructureReady: true`, watch `OpenShiftControlPlane.Status` until `Initialized: true`
Then it goes to watching clusteroperators in the way it already does.

So I'd expect the installer to start interpreting the conditions on these two objects and reporting the errors when they change. In my experience they are pretty good at updating the status.

That said, we can also stream the logs from the controllers in a debug mode if we wanted to. Perhaps rather than dumping to the terminal output, we could capture the CAPI logs into a debug file/folder, which I think we already do for terraform